spring-web/src/main/java/org/springframework/http/codec/json/AbstractJackson2Decoder.java

```diff
@@ -174,22 +177,38 @@ protected Flux<DataBuffer> processInput(Publisher<DataBuffer> input, ResolvableT
 	@Override
 	public Mono<Object> decodeToMono(Publisher<DataBuffer> input, ResolvableType elementType,
 			@Nullable MimeType mimeType, @Nullable Map<String, Object> hints) {
-
 		return DataBufferUtils.join(input, this.maxInMemorySize)
-				.flatMap(dataBuffer -> Mono.justOrEmpty(decode(dataBuffer, elementType, mimeType, hints)));
+				.flatMap(dataBuffer -> {
+							try {
+								ObjectReader objectReader = getObjectReader(elementType, mimeType, hints);
+								return customizeReaderFromStream(objectReader, mimeType, elementType, hints)
+										.flatMap(reader -> {
+											try {
+												Object result = decode(dataBuffer, reader, hints);
+												return Mono.just(result);
```

Comment:
Did you purposely use Mono.just over the original Mono.justOrEmpty?

Reply:
No, good catch, this should be Mono.justOrEmpty.  I'll fix it, thanks